backend/cfg/cfgize.ml

```diff
@@ -459,6 +415,33 @@ let rec add_blocks :
       | Cfg.Tailcall _ | Cfg.Call_no_return _ ->
         body
     in
+    let can_raise =
+      (* Recompute [can_raise] and check that instructions in the middle of the
+         block do not raise, i.e., only the terminator, or the last instruction
+         (when the terminator is just a goto) can raise. *)
+      let terminator_is_goto =
+        match (terminator.Cfg.desc : Cfg.terminator) with
+        | Always _ -> true
+        | Raise _
+        | Tailcall (Func _)
+        | Call_no_return _ | Never | Parity_test _ | Truth_test _ | Float_test _
+        | Int_test _ | Switch _ | Return
+        | Tailcall (Self _) ->
```

Comment:
There is no need to have two `Tailcall` patterns.

Reply:
fixed